### PR TITLE
CPP: Fix performance for cpp/cleartext-transmission

### PR DESCRIPTION
**Pull request by geoffw0**
There was a moderate performance explosion on some projects (such as `nlohmann__json`), caused by the `Encrypted` class.  Tested locally, DCA run to follow.

### CODE REVIEW

File: cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql [hunk 1]
```diff
@@ -168,6 +168,14 @@ class NetworkRecv extends NetworkSendRecv {
   override Recv target;
 }
 
+predicate encryptionFunction(Function f) {
+  f.getName().toLowerCase().regexpMatch(".*(crypt|encode|decode|hash|securezero).*")
+}
+
+predicate encryptionType(UserType t) {
+  t.getName().toLowerCase().regexpMatch(".*(crypt|encode|decode|hash|securezero).*")
+}
```
Comment (MathiasVP) [inline cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql:177]: If this is for a join-order fix you probably want to mark them as `pragma[noinline]`.

Comment (geoffw0) [reply cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql]: Done (and improves performance a bit further)!

Comment (MathiasVP) [reply cpp/ql/src/Security/CWE/CWE-311/CleartextTransmission.ql]: Perfect! Do you happen to have tuple counts before and after this change?

**Comment by geoffw0** [comment]
DCA shows a significant speedup and no change in results. :)

**Pull request merged**